crates/matrix-sdk-crypto/src/olm/account.rs

```diff
@@ -424,6 +431,11 @@ impl Account {
         Self::new_helper(account, user_id, &device_id)
     }
 
+    /// Mark the account as being used for a dehydrated device.
+    pub(crate) fn mark_as_dehydrated(&mut self) {
```

Comment:
I think that this should be a constructor. We don't want to have the ability to convert an existing account into a `Dehydrated` one, we want to either create a normal `Account` or for dehydration.

The `DehydratedDevices::create()` method then creates a dehydrated one.